rmgpy/rmg/main.py

```diff
@@ -1463,21 +1481,126 @@ def generateCanteraFiles(self, chemkinFile, **kwargs):
+
+                for index, adj_list in restartSpeciesMap.items():
+                    restartSpeciesMap[index] = Species().fromAdjacencyList(adj_list)
+
+                # Load in the restart filter tensors
+                with h5py.File(self.filtersPath, 'r') as f:
+                    try:
+                        unimolecularThreshold_restart = f.get('unimolecularThreshold').value
+                        bimolecularThreshold_restart = f.get('bimolecularThreshold').value
+                        if self.trimolecular:
+                            trimolecularThreshold_restart = f.get('trimolecularThreshold').value
+
+                        # Expand Thresholds to match number of species in the current model.
+                        unimolecularThreshold = np.zeros(numCoreSpecies, bool)
+                        unimolecularThreshold[:numRestartSpcs] = unimolecularThreshold_restart
+
+                        bimolecularThreshold = np.zeros((numCoreSpecies, numCoreSpecies), bool)
+                        bimolecularThreshold[:numRestartSpcs, :numRestartSpcs] = bimolecularThreshold_restart
+
+                        if self.trimolecular:
+                            trimolecularThreshold = np.zeros((numCoreSpecies, numCoreSpecies, numCoreSpecies), bool)
+                            trimolecularThreshold[:numRestartSpcs, :numRestartSpcs, :numRestartSpcs] = \
+                                trimolecularThreshold_restart
+
+                        filters_found = True
+
+                    except KeyError:  # If we can't find the filters then this is because filtering was not used
+                        logging.warning('No filters were found in file {0}. This is to be expected if the restart '
+                                        'files specified came from an RMG job that did not use filtering. Therefore, '
+                                        'RMG will assume that all of the core species from the restart seed have '
+                                        'already been reacted and will not react them again. Additional species that '
+                                        'were added to the input file but are not in the restart core seed WILL be '
+                                        'reacted with the rest of the core though. If this is not the intended '
+                                        'behavior please stop this job and correct the missing '
+                                        'filters.'.format(self.filtersPath))
+
+                        filters_found = False
+
+                # Reorder the core species to match the indices of the restart filter tensors
+                core_species_map = {}
+
+                for index, spcs in restartSpeciesMap.items():
+                    for j, core_spcs in enumerate(self.reactionModel.core.species):
+                        if core_spcs.isIsomorphic(spcs):
+                            core_species_map[index] = core_spcs
```

Comment:
You could do `core_species_map[index] = self.reactionModel.core.species.pop(j)` instead of doing it after.

Since we break immediately after, it doesn't matter if we shorten the list.

Also, I still think that the restart species map can just be dumped as a list instead of a dictionary with indices as keys.

Reply:
Good point, I'll do this